web/app_options.js

```diff
@@ -13,13 +13,110 @@
  * limitations under the License.
  */
 
+import { viewerCompatibilityParams } from './viewer_compatibility';
+
 const OptionKind = {
   VIEWER: 'viewer',
   API: 'api',
   WORKER: 'worker',
 };
 
 const defaultOptions = {
+  defaultZoomValue: {
+    /** @type {string} */
+    value: '',
+    kind: OptionKind.VIEWER,
+  },
+  disableFullscreen: {
+    /** @type {boolean} */
+    value: viewerCompatibilityParams.disableFullscreen || false,
+    kind: OptionKind.VIEWER,
+  },
+  disableHistory: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  disablePageLabels: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  disablePageMode: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  enablePrintAutoRotate: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  enableWebGL: {
+    /** @type {boolean} */
+    value: false,
+    kind: OptionKind.VIEWER,
+  },
+  externalLinkRel: {
+    /** @type {string} */
+    value: null,
+    kind: OptionKind.VIEWER,
+  },
+  externalLinkTarget: {
+    /** @type {number} */
+    value: 0,
+    kind: OptionKind.VIEWER,
+  },
+  imageResourcesPath: {
+    /** @type {string} */
+    value: './images/',
+    kind: OptionKind.VIEWER,
+  },
+  locale: {
+    /** @type {string} */
+    value: (typeof navigator !== 'undefined' ? navigator.language : 'en-US'),
+    kind: OptionKind.VIEWER,
+  },
+  maxCanvasPixels: {
+    /** @type {number} */
+    value: viewerCompatibilityParams.maxCanvasPixels || 16777216,
```

Comment:
I'd very much prefer to *not* do that, for a couple of reasons.

First of all, the "source of truth" in the new code is the JSDocs added to the viewer components and API methods, rather than this file itself; again compare with https://github.com/mozilla/pdf.js/blob/master/web/default_preferences.json.
In this particular case, the documentation is already present in https://github.com/mozilla/pdf.js/blob/5f98f9b8f10d68df6cbd331b5e135e167c71829b/web/base_viewer.js#L59-L61 and https://github.com/mozilla/pdf.js/blob/5f98f9b8f10d68df6cbd331b5e135e167c71829b/web/base_viewer.js#L59-L61

Second of all, if we start adding that kind of comments in this file we'll probably end up doing it for *a lot* more options. That would add maintenance burden, since each JSDoc comment would then be duplicated twice in the code-base. Furthermore, this file would become much longer and quite unwieldy in my opinion.